firmware/Core/Src/rtos_tasks/rtos_tasks.c

```diff
@@ -72,6 +72,7 @@ void TASK_handle_uart_telecommands(void *argument) {
 			
 			// copy the buffer to the latest_tcmd buffer
 			latest_tcmd_len = UART_telecommand_buffer_write_idx;
+			// FIXME: rewrite the following memcpy/memset without the casts (which disregards the volatile qualifier)
```

Comment:
Any idea of the significance of this FIXME, or how it might be done?

Reply:
You can't call memcpy nor memset with `volatile uint8_t*` (nor char) args because the implementation of those functions may use multi-byte operations under the hood. As a hack, I simply type-casted to make them non-volatile, but the correct approach is to use a for-loop.

The idea is that in the very-rare case that the ISR that writes to the `UART_...buffer` variable is called during this clearing operation, doing a mem-set is a for-loop is at least defined behaviour. 

Hardly matters since no UART telecommands will be issued in space, but type casts in C should be used sparingly. In case anyone uses with as an example, it'd be ideal to not cast

Reply:
Thanks!